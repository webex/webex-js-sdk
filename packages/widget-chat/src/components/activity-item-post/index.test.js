import React from 'react';
import renderer from 'react-test-renderer';

import ActivityItemPost from '.';

describe(`ActivityItemPost post component`, () => {
  const activity = {
    id: `test-123-123-123-123`,
    content: `Test Activity Content`,
    name: `Test User`,
    timestamp: `2016-09-20T19:52:57.186Z`,
    verb: `post`
  };

  const component = renderer.create(
    <ActivityItemPost
      content={activity.content}
      id={activity.id}
      name={activity.name}
      timestamp={activity.timestamp}
      verb={activity.verb}
    />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});
