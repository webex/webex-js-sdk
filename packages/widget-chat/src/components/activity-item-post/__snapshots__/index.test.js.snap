exports[`ActivityItemPost post component renders properly 1`] = `
<div
  className="activity-item">
  <div
    className="avatar-wrapper">
    <div
      className="avatar"
      style={undefined}>
      <span
        className="avatar-self" />
    </div>
  </div>
  <div
    className="content-container">
    <div
      className="meta">
      <div
        className="display-name"
        title="{name}">
        Test User
      </div>
      <div
        className="published">
        2016-09-20T19:52:57.186Z
      </div>
    </div>
    <div
      className="activity-content">
      <div
        className="activity-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Test Activity Content",
          }
        } />
    </div>
  </div>
  <div
    className="activity-post-actions">
    <div
      className="activity-post-action">
      <div
        className="post-action-item">
        <button
          className="button"
          onClick={[Function]}
          onKeyUp={[Function]}>
          <div
            className="icon"
            title="Flag this message" />
        </button>
      </div>
    </div>
    <div
      className="activity-post-action">
      <div
        className="post-action-item">
        <button
          className="button"
          onClick={[Function]}
          onKeyUp={[Function]}>
          <div
            className="icon"
            title="Delete this message" />
        </button>
      </div>
    </div>
  </div>
</div>
`;
